class.jetpack-admin.php

```diff
@@ -51,6 +52,39 @@ private function __construct() {
 
 		// Add module bulk actions handler
 		add_action( 'jetpack_unrecognized_action', array( $this, 'handle_unrecognized_action' ) );
+
+		// If the site has Jetpack Anti-Spam, change the Akismet menu label accordingly.
+		$site_products = Jetpack_Site_Products::get();
+		if ( false !== array_search( 'jetpack_anti_spam', array_column( $site_products, 'product_slug' ), true ) ) {
+			// Prevent Akismet from adding a menu item.
+			add_action(
+				'init',
+				function () {
+					remove_action( 'admin_menu', array( 'Akismet_Admin', 'admin_menu' ), 5 );
+				},
+				99
```

Comment:
This piece of code came from a suggestion I received in `pZbK5-5Rp-p2`.

I'm not very familiar with how the hooks system works, but I tested a couple of options and they didn't work (I probably did something wrong). This is what I tried.

Option 1:
```
add_action(
	'jetpack_admin_menu',
	function () {
		remove_action( 'admin_menu', array( 'Akismet_Admin', 'admin_menu' ), 5 );
	},
	99
);
```

Option 2:
```
add_action(
	'admin_menu',
	function () {
		remove_action( 'admin_menu', array( 'akismet_admin', 'admin_menu' ), 15 );
	},
	99
);
```

In both cases, both menu items were visible (Akismet and Jetpack Anti-Spam).

Reply:
Akismet adds another hook after the `admin_menu` call, so we could use this code:

```
add_action(
	'jetpack_admin_menu',
	function () {
		remove_action( 'jetpack_admin_menu', array( 'Akismet_Admin', 'load_menu' ), 5 );
	},
	5
);
```
We have to remove the hook that calls `load_menu` and set the priority higher (5) than the default (10). So with that, we ensure that the call to the hook happens only in the admin pages instead of the whole of the WordPress site.

Reply:
Instead of `priority`=99 in option 2, I think you'd want `priority`=4.  And I agree that doing it that way is better than hooking into `init`.

Reply:
And the priority needs to match. So it should be: 

```
add_action(
	'admin_menu',
	function () {
		remove_action( 'admin_menu', array( 'akismet_admin', 'admin_menu' ), 5 );
	},
	4
);
```

(untested)

Reply:
This ^ one worked well so I updated the PR to rely on `admin_menu` instead of `init`. Thank you both!